tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,343 @@
+                self.customer_df,
+                event_timestamp_column="event_timestamp",
+                created_timestamp_column="created",
+            )
+            self._customer_feature_view = create_customer_daily_profile_feature_view(ds)
+        return self._customer_feature_view
+
+    def driver_stats_feature_view(self) -> FeatureView:
+        if self._driver_stats_feature_view is None:
+            driver_table_id = self.data_source_creator.get_prefixed_table_name(
+                self.name, "driver_hourly"
+            )
+            ds = self.data_source_creator.create_data_source(
+                driver_table_id,
+                self.driver_df,
+                event_timestamp_column="event_timestamp",
+                created_timestamp_column="created",
+            )
+            self._driver_stats_feature_view = create_driver_hourly_stats_feature_view(
+                ds
+            )
+        return self._driver_stats_feature_view
+
+    def orders_table(self) -> Optional[str]:
+        if self._orders_table is None:
+            orders_table_id = self.data_source_creator.get_prefixed_table_name(
+                self.name, "orders"
+            )
+            ds = self.data_source_creator.create_data_source(
+                orders_table_id,
+                self.orders_df,
+                event_timestamp_column="event_timestamp",
+                created_timestamp_column="created",
+            )
+            if hasattr(ds, "table_ref"):
+                self._orders_table = ds.table_ref
+            elif hasattr(ds, "table"):
+                self._orders_table = ds.table
+        return self._orders_table
+
+
+def vary_full_feature_names(configs: List[TestRepoConfig]) -> List[TestRepoConfig]:
+    new_configs = []
+    for c in configs:
+        true_c = replace(c, full_feature_names=True)
+        false_c = replace(c, full_feature_names=False)
+        new_configs.extend([true_c, false_c])
+    return new_configs
+
+
+def vary_infer_event_timestamp_col(
+    configs: List[TestRepoConfig],
+) -> List[TestRepoConfig]:
+    new_configs = []
+    for c in configs:
+        true_c = replace(c, infer_event_timestamp_col=True)
+        false_c = replace(c, infer_event_timestamp_col=False)
+        new_configs.extend([true_c, false_c])
+    return new_configs
+
+
+def vary_providers_for_offline_stores(
+    configs: List[TestRepoConfig],
+) -> List[TestRepoConfig]:
+    new_configs = []
+    for c in configs:
+        if "FileDataSourceCreator" in c.offline_store_creator:
+            new_configs.append(c)
+        elif "RedshiftDataSourceCreator" in c.offline_store_creator:
+            for p in ["local", "aws"]:
+                new_configs.append(replace(c, provider=p))
+        elif "BigQueryDataSourceCreator" in c.offline_store_creator:
+            for p in ["local", "gcp"]:
+                new_configs.append(replace(c, provider=p))
+    return new_configs
+
+
+@contextmanager
+def construct_test_environment(
```

Comment:
Calling the methods https://github.com/woop/feast-test-review/pull/1/files#r697784987 from this function makes more sense to me. Also, this function should probably be renamed to something that makes it clear it's not for general use but instead for constructing the a universal repository.